### DISCUSSION TITLE
Does ecode use Lua internally?

### DISCUSSION THREAD

**Discussion by ghost** [SpartanJ/ecode#143] [General]
Or the only Lua scripts are premake5 build scripts? I found Lua Pattern in Find/Replace. It seems Lua is there. But only not as visible as in Lite XL and Pragtical.

**Comment by emeraldtip** [comment]
It's c++.
https://github.com/SpartanJ/eepp/tree/develop/src/tools/ecode

**Comment by SpartanJ** [comment]
It's pure C++. No Lua usage currently (that might change in the future).

**Reply by ghost** [reply]
How do you support Lua Pattern without Lua itself?

**Reply by SpartanJ** [reply]
Lua is implemented in C, so I took the Lua patterns implementation which is extremely simple, this is why I love it. [Here's](https://github.com/SpartanJ/eepp/blob/develop/src/eepp/system/lua-str.cpp) the implementation. Lua patterns are *really* fast.

**Reply by ghost** [reply]
> Lua is implemented in C, so I took the Lua patterns implementation which is extremely simple

Can you do the same with LPeg?

https://www.inf.puc-rio.br/~roberto/lpeg/

**Reply by SpartanJ** [reply]
I think it's the same answer than [here](https://github.com/SpartanJ/ecode/issues/176#issuecomment-1979876104). The problem of LPeg is that it must be executed in a Lua environment or at least in some programming language (I guess you could port the library with the operators overloading to other languages) which makes it very limited, it's also not widely used, so I don't see a great benefit in implementing it, it makes much more sense just to add PCRE support (which is going to happen).
